typespec-tests/src/test/java/com/type/property/additionalproperties/SpreadTests.java

```diff
@@ -0,0 +1,301 @@
+
+        DifferentSpreadStringRecord record = spreadDifferentStringClient.get();
+        Assertions.assertNotNull(record);
+        Assertions.assertEquals(43.125, record.getId());
+        Assertions.assertNotNull(record.getAdditionalProperties());
+        Assertions.assertEquals("abc", record.getAdditionalProperties().get("prop"));
+    }
+
+    @Test
+    public void testSpreadDifferentFloat() {
+        Map<String, Double> propertyMap = new LinkedHashMap<>();
+        propertyMap.put("prop", 43.125);
+        DifferentSpreadFloatRecord body = new DifferentSpreadFloatRecord("abc");
+        body.setAdditionalProperties(propertyMap);
+        spreadDifferentFloatClient.put(body);
+
+        DifferentSpreadFloatRecord record = spreadDifferentFloatClient.get();
+        Assertions.assertNotNull(record);
+        Assertions.assertEquals("abc", record.getName());
+        Assertions.assertNotNull(record.getAdditionalProperties());
+        Assertions.assertEquals(43.125, record.getAdditionalProperties().get("prop"));
+    }
+
+    @Test
+    public void testSpreadDifferentModel() {
+        Map<String, ModelForRecord> propertyMap = new LinkedHashMap<>();
+        propertyMap.put("prop", new ModelForRecord("ok"));
+        DifferentSpreadModelRecord body = new DifferentSpreadModelRecord("abc");
+        body.setAdditionalProperties(propertyMap);
+        spreadDifferentModelClient.put(body);
+
+        DifferentSpreadModelRecord record = spreadDifferentModelClient.get();
+        Assertions.assertNotNull(record);
+        Assertions.assertEquals("abc", record.getKnownProp());
+        Assertions.assertNotNull(record.getAdditionalProperties());
+        Assertions.assertNotNull(record.getAdditionalProperties().get("prop"));
+        Assertions.assertEquals("ok", record.getAdditionalProperties().get("prop").getState());
+    }
+
+    @Test
+    @Disabled("The 'Get' request did not respond for a long time.")
+    public void testSpreadDifferentModelArray() {
+        Map<String, List<ModelForRecord>> propertyMap = new LinkedHashMap<>();
+        propertyMap.put("prop", Arrays.asList(new ModelForRecord("ok"), new ModelForRecord("ok")));
+        DifferentSpreadModelArrayRecord body = new DifferentSpreadModelArrayRecord("abc");
+        body.setAdditionalProperties(propertyMap);
+        spreadDifferentModelArrayClient.put(body);
+
+        DifferentSpreadModelArrayRecord record = spreadDifferentModelArrayClient.get();
+        Assertions.assertNotNull(record);
+        Assertions.assertEquals("abc", record.getKnownProp());
+        Assertions.assertNotNull(record.getAdditionalProperties());
+        Assertions.assertNotNull(record.getAdditionalProperties().get("prop"));
+        record.getAdditionalProperties().get("prop").forEach(modelForRecord ->
+                Assertions.assertEquals("ok", modelForRecord.getState()));
+    }
+
+    @Test
+    @Disabled("The body provided of 'put' request doesn't match expected body")
```

Comment:
What's wrong with this test?

Reply:
Enabled the test cases except `*ModelArray`.